### ISSUE TITLE
[attributes-order] angular input binding without brackets considered as attribute binding

### ISSUE THREAD

**Issue by joeldhenry** [angular-eslint/angular-eslint#1729]
**Description and reproduction of the issue**
a bug with template lint. attribute order doesnt understand what is attribute or input. angular input references such as `name="title"` should be ordered separately to html attributes such as `class="my-object`. noted a feature to consider classes as seperate ordering entity also.

```JSON
{
    "@angular-eslint/template/attributes-order": [
    "error",
    {
      "alphabetical": false,
      "order": [
        ...
        "INPUT_BINDING",
        "ATTRIBUTE_BINDING",
        "CLASS"
      ]
    }
  ]
}
```

```HTML
<ng-icon class="test" name="preview" /> <-- should error - name is angular input
<ng-icon name="preview" class="test" /> <-- good
```

**Versions**

Angular CLI: 16.1.6
Node: 18.18.0
Package Manager: npm 9.8.1
OS: darwin arm64

Angular: 16.1.7
... animations, common, compiler-cli, core, language-service
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1601.6
@angular-devkit/build-angular   16.1.6
@angular-devkit/core            16.1.6
@angular-devkit/schematics      16.1.6
@angular/cli                    16.1.6
@schematics/angular             16.1.6
rxjs                            7.8.1
typescript                      5.1.6
zone.js                         0.13.1

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by joeldhenry** [comment]
similar to https://github.com/angular-eslint/angular-eslint/issues/1267 - but i would consider this a bug rather than a feature improvement

**Comment by reduckted** [comment]
The problem is that the Angular template parser says that `class` and `name` are both attributes. It only says that things are inputs when they have square brackets. Unfortunately, I don't think there's anything that can be done about this.

**Comment by JamesHenry** [comment]
Sorry for the delay on this one! (And great surname, Joel 😉)

Indeed, they are syntactically identical and cannot be differentiated without semantic information from the Angular compiler (that is not available to us).

I'll close this one as it's not something we can do
